src/com/xilinx/rapidwright/rwroute/RouteNode.java

```diff
@@ -451,47 +453,48 @@ public void resetChildren() {
     }
 
     private void setType(RouteNodeType type) {
-        this.type = type;
+        RouteNodeType typeToSet = type;
         if (type == RouteNodeType.WIRE) {
             // NOTE: IntentCode is device-dependent
             IntentCode ic = node.getIntentCode();
             switch (ic) {
                 case NODE_PINBOUNCE:
-                    this.type = RouteNodeType.PINBOUNCE;
+                    typeToSet = RouteNodeType.PINBOUNCE;
```

Comment:
I would get rid of `typeToSet` entirely and overwrite `type` for clarity. It looks like we don't read from `type` except in the very first `if`?